### ISSUE TITLE
Editor stripping code when in page editor

### ISSUE THREAD

**Issue by KwrKy** [benjaminprojas/wp-editor#72]
Hi Benajmin

I really like the plugin, and am quite surprised that there aren't more like it. I only need it for working with pages and posts, but I can't use it at the moment for a couple of reasons:

**1. CRITICAL: It is stripping code.** 

The code in the original text editor was

```
<button class="downbuttonabout">
<a href="#downabout"><h4 style="color:#fff;">
<span id="downbutton-text">About Us</span>
</h4>
</a>
```

When I activated your editor, it became
```
<h4 style="color: #fff;">
<span id="downbutton-text">About Us</span>
</h4>
```

I know that the Wordpress TinyMCE strips <span> tags with nothing between them, but the fact that it stripped both the button tag and the hyperlink is very concerning. That wasn't the only place it stripped code, just the best example for this comment.

**2. INTENSELY FRUSTRATING: The syntactic indentation is all over the place**

The first level indent is as it is set, but the second level (and further) suddenly jumps halfway across the screen. Also, some nested divs spontaneously decide to start from the beginning again. It's really chaotic and hard to follow.
![screen shot 2017-02-16 at 12 57 07](https://cloud.githubusercontent.com/assets/25687099/23018660/7840cf94-f447-11e6-943c-64c9d10c877c.png)

I'd love to use your plugin because it solves a major headache, but I just can't trust it at the moment. If you can solve those issues I'd hop right back on 👍 

Thanks for the great (if still rather buggy) plugin.

Peace

**Comment by YellowAfterlife** [comment]
The indentation issue comes from CodeMirror itself. You can notice it on majority of websites and tools using the library.

For a while now I've been using a small userscript that does
```js
wp_editor.setOption("smartIndent", false);
```
on page load.
